### PR TITLE
MySQL8 support

### LINKED ISSUE #4345: Database schema is incompatible with utf8, the default in MySQL 8
As found out in #2478, the [`mutation_event`](https://github.com/cBioPortal/cbioportal/blob/v1.13.2/db-scripts/src/main/resources/cgds.sql#L395) table contains a unique key too large to be accepted by MySQL when using the UTF-8 character set.

Since this is the [default encoding](https://dev.mysql.com/doc/refman/8.0/en/charset-applications.html) in the current release of MySQL, I have documented a recommendation to stick to MySQL 5.7 in #4333, along with a hint at why in case users do want to tweak the configuration.

However, I feel that we should adjust the schema to support UTF-8 and MySQL 8.x out of the box.

### PR DESCRIPTION
**Pull request by khzhu**
# What? Why?

- Adjust table column size to 255 bytes to be compatible with UTF8 
- Limit 3072 bytes per index to support MySQL8
- Fix #4345

Changes proposed in this pull request:
- Change **mutation_event** table schema
- - Downsize columns with size > 255 to 255 bytes
- - Limit the unique constraint to 3072 bytes
- Upgrade **mysql-connector-java** to the latest `8.0.11`
- Update JDBC driver from `com.mysql.jdbc.Driver` to `com.mysql.cj.jdbc.Driver`
- Add a new `db.timezone` property to portal.properties file 

# Checks
- [ ] Runs on Heroku.
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Google Style Guide](https://github.com/google/styleguide).
- [ ] If this is a feature, the PR is to rc. If this is a bug fix, the PR is to master.

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

### CODE REVIEW

**Comment by fedde-s** [comment]
Is this still waiting for review?

**Comment by khzhu** [comment]
Hi @fedde-s , I am not sure the plan. Hi @jjgao , can you please let us know the timeline to update cbioportal to mysql8? thanks!

**Comment by jjgao** [comment]
@khzhu I think we should try to get this reviewed and merged. Could you work with @n1zea144 and @sheridancbio on this? @fedde-s I am wondering if you can also help review?

**Comment by khzhu** [comment]
@jjgao , It has been really a while. I will get a PR updated before asking Ben's team for the review.

**Comment by jjgao** [comment]
Thanks, @khzhu. Please feel free to bug the reviewers after you update the pr :)

**Comment by jjgao** [comment]
@khzhu there are so many conflicts. I'll close this PR for now. Let's keep the issue on the back burner for now.

File: db-scripts/src/main/resources/migration.sql [hunk 1]
```diff
@@ -454,3 +454,13 @@ ALTER TABLE gistic_to_gene DROP FOREIGN KEY gistic_to_gene_ibfk_2;
 ALTER TABLE gistic_to_gene ADD CONSTRAINT `gistic_to_gene_ibfk_2` FOREIGN KEY (`GISTIC_ROI_ID`) REFERENCES `gistic` (`GISTIC_ROI_ID`) ON DELETE CASCADE;
 
 UPDATE info SET DB_SCHEMA_VERSION="2.6.0";
+
+##version: 2.7.0
+-- adjust mutation_event table columns to be compatible with UTF8, the default in mySQL8
```
Comment (pieterlukasse) [inline db-scripts/src/main/resources/migration.sql]: I would just add the script here.

**Comment by pieterlukasse** [comment]
The PR should also update the Docker files where necessary.

**Pull request closed without merging**